internal/services/containers/kubernetes_cluster_resource.go

```diff
@@ -2158,6 +2157,10 @@ func resourceKubernetesClusterUpdate(d *pluginsdk.ResourceData, meta interface{}
 			ebpfDataPlane := d.Get(key).(string)
 			existing.Model.Properties.NetworkProfile.NetworkDataplane = pointer.To(managedclusters.NetworkDataplane(ebpfDataPlane))
 		}
+
+		if key := "network_profile.0.outbound_type"; d.HasChange(key) {
+			existing.Model.Properties.NetworkProfile.OutboundType = pointer.To(managedclusters.OutboundType(d.Get(key).(string)))
```

Comment:
we also need to reset LoadBalancerProfile if outbound type is not loadbalancer and rest NATGatewayProfile if outbound type is not managedNATGateway

Reply:
@MartinForReal feel free to open a PR, please make sure to include a test case to demonstrate the use case that is meant to be supported. If you're unable to I'd recommend opening an issue with the details so that maintainers or members in the community can take a look into this if there's free bandwidth.